src/pathfinder/yapf/yapf_ship.cpp

```diff
@@ -21,7 +21,7 @@
 
 constexpr int NUMBER_OR_WATER_REGIONS_LOOKAHEAD = 4;
 constexpr int MAX_SHIP_PF_NODES = (NUMBER_OR_WATER_REGIONS_LOOKAHEAD + 1) * WATER_REGION_NUMBER_OF_TILES * 4; // 4 possible exit dirs per tile.
-
+constexpr int SHIP_REVERSAL_PENALTY = YAPF_TILE_LENGTH * 10;
```

Comment:
```suggestion
constexpr int SHIP_REVERSAL_PENALTY = YAPF_TILE_LENGTH * 25;
```
I tested values as low as 1500 and 2000, and they were still causing issues in the tests. 2500 was the lowest value I tested which doesn't cause ships to be going back and forth.

Reply:
How big is the loop that the ships that are still going in circles are making with the 2000 penalty?
Am I right that those "circles" are mostly up/down or left/right on the map, i.e. over the diagonals?

Might then be that the minimum penalty is related to the maximum distance within a tile, e.g. 32 "half" tiles, with a penalty of 32 * YAPF_TILE_CORNER_LENGTH (71) => 2272. That could explain why 2000 doesn't work and 2500 does. Though this is pure speculation from my part. 
It would be nice if there were some background to why a particular penalty was chosen. As "arbitrary" by "empirical research" might not be the best reason.